### PR TITLE
Circulation: invenio-circulation integration

### PR DESCRIPTION
**Pull request by BadrAly**
NEW: circulation apis refactoring

Co-Authored-by: Aly Badr aly.badr@rero.ch
Co-Authored-by: Gianni Pante gianni.pante@rero.ch

To Test:

- get the PR branch
- ./scripts/bootstrap
- ./scripts/setup
- ./run_tests.sh

application_test:
Go through all the functionalities you can on the localhost:5000 except the circulation_ui.

### CODE REVIEW

File: rero_ils/modules/documents_items/cli.py [hunk 1]
```diff
@@ -90,38 +87,45 @@ def create_items(verbose, count, itemscount, missing,
...
-        'item_type_pid': random.choice(item_types_pids)
+        'item_type_pid': random.choice(item_types_pids),
     }
     item = Item.create(data)
 
     n = int(item.pid)
     data['barcode'] = str(10000000000 + n)
     data['call_number'] = str(n).zfill(5)
     item.update(data)
+    document.add_item(item, dbcommit=True)
```
Comment (rerowep) [inline rero_ils/modules/documents_items/cli.py]: No reindex=True ???

File: rero_ils/modules/documents_items/cli.py [hunk 2]
```diff
@@ -143,15 +147,15 @@ def create_loan(patron_barcode, library_pid, short):
...
     else:
-        end = 45-n
+        end = 45 - n
     end_date = (current_date + datetime.timedelta(days=end)).isoformat()
     request = {
         'patron_barcode': patron_barcode,
         'pickup_library_pid': library_pid,
         'start_date': start_date,
-        'end_date': end_date
+        'end_date': end_date,
```
Comment (rerowep) [inline rero_ils/modules/documents_items/cli.py]: , at the end of dict not necessary but ok

File: rero_ils/modules/items/cli.py [hunk 3]
```diff
@@ -98,8 +101,11 @@ def create_circ_transactions(infile, verbose):
 
 def print_message(barcode, item_barcode, transaction_type):
     """Print confirmation message."""
-    click.echo('\t {0} created for patron {1} and item {2}'
-               .format(transaction_type, barcode, item_barcode))
+    click.echo(
+        '\t {0} created for patron {1} and item {2}'.format(
```
Comment (rerowep) [inline rero_ils/modules/items/cli.py]: not from you, but better to write 
`"{transaction_type}".format(transaction_type=transaction_type)`

File: rero_ils/modules/documents_items/cli.py (hunk 1 above)
Comment (BadrAly) [reply rero_ils/modules/documents_items/cli.py]: indexing @ line #94

File: rero_ils/utils.py [hunk 4]
```diff
@@ -65,6 +65,16 @@ def i18n_to_str(language):
 def resolve_function(function_name):
     """Execute resolve function."""
     result = []
+    if function_name == 'libraries_pids_names':
+        from rero_ils.modules.libraries_locations.api import Library
+        for id in Library.get_all_ids():
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_ils/utils.py]: why not library_id

File: rero_ils/config.py [hunk 5]
```diff
@@ -931,22 +968,17 @@ def _(x):
...
-]
+
 CIRCULATION_ITEM_LOCATION_RETRIEVER = Item.item_location_retriever
-CIRCULATION_DOCUMENT_RETRIEVER_FROM_ITEM = (
+CIRCULATION_DOCUMENT_RETRIEVER_FROM_ITEM = \
     DocumentsWithItems.document_retriever
-)
 CIRCULATION_ITEMS_RETRIEVER_FROM_DOCUMENT = DocumentsWithItems.items_retriever
 
+SERVER_NAME = 'localhost:5000'
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ils/config.py]: needed to specify the port in the URL (url_for)

File: rero_ils/modules/items/cli.py [hunk 6]
```diff
@@ -126,36 +150,76 @@ def get_loan_dates(transaction_type, item):
...
+    )
+    for patron in patrons:
+        record = patron.to_dict()
+        patron_pid = record.get('pid')
+        return Patron.get_record_by_pid(patron_pid)
+
+
+def get_one_user_data():
+    """Find a qualified user data."""
+    user = get_one_staff_user().dumps()
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ils/modules/items/cli.py]: needed to get the circulation_location

**Pull request merged**